ts/client/features/deposit.ts

```diff
@@ -184,7 +175,64 @@ export class DepositHandlingStrategy implements BatchHandlingStrategy {
 }
 
 export class DepositPackingCommand implements BatchPackingCommand {
-    async packAndSubmit(): Promise<ContractTransaction> {
-        throw new Error("Not implemented");
+    constructor(
+        private params: DeploymentParameters,
+        private storageManager: StorageManager,
+        private pool: IDepositPool,
+        private rollup: Rollup
+    ) {}
+
+    private async submitDeposits(
+        prevBatch: Batch,
+        vacancy: Vacant
+    ): Promise<ContractTransaction> {
+        const prevCommitProof = prevBatch.proofCompressed(
+            prevBatch.commitments.length - 1
+        );
+
+        return await this.rollup.submitDeposits(prevCommitProof, vacancy, {
+            value: this.params.STAKE_AMOUNT
+        });
+    }
+
+    public async packAndSubmit(): Promise<ContractTransaction> {
+        const { state, batches } = this.storageManager;
+
+        const curBatch = await this.storageManager.batches.current();
+        if (!curBatch) {
+            throw new Error("no batches synced");
+        }
+
+        const vacancy = await this.storageManager.state.findVacantSubtree(
+            this.params.MAX_DEPOSIT_SUBTREE_DEPTH
+        );
+
+        console.log("Submitting deposits", prettyVacant(vacancy));
+        const l1Txn = await this.submitDeposits(curBatch, vacancy);
+
+        // Update L2 state
+        const subtree = this.pool.popDepositSubtree();
+        await state.updateBatch(
+            vacancy.pathAtDepth,
+            this.params.MAX_DEPOSIT_SUBTREE_DEPTH,
+            subtree.states
+        );
+        await state.commit();
+
+        const context = {
+            /*
+             * TODO PR Comment Why do we need the subtreeID in this context?
+             * It is not currently being used anywhere in the client.
+             */
+            subtreeID: -1,
```

Comment:
Feedback and thoughts appreciated on this.

Reply:
Good question. The subtreeID is the reliable unique identifier for the deposit subtree. We can work without subtreeID in the happy case, but when we need to debug, handle reorg, present deposits in logs or a dashboard, the subtreeID would come in handy.

(The subtreeID was proposed in #472)

Reply:
Also validate the subtreeId is valid